### PR TITLE
api: switch to IndexDB

### PR DESCRIPTION
**Pull request by dcposch**
Decoupling `daimo-api` from indexer implementation details, using #1239

### CODE REVIEW

File: packages/daimo-api/src/contract/SwapClogMatcher.ts [hunk 1]
```diff
@@ -20,39 +23,56 @@ export class SwapClogMatcher {
...
         blockHash: bytesToHex(row.block_hash, { size: 32 }),
         blockNumber: BigInt(row.block_num),
         transactionHash: bytesToHex(row.tx_hash, { size: 32 }),
-        transactionIndex: row.tx_idx,
-        logIndex: row.log_idx,
-        address: getAddress(bytesToHex(row.log_addr, { size: 20 })),
+        transactionIndex: Number(row.tx_idx),
+        logIndex: Number(row.sort_idx),
+        address:
+          row.token == null
```
Comment (andrewliu08) [inline packages/daimo-api/src/contract/SwapClogMatcher.ts:68]: in s^2, should we fill the token to zero address instead of having it as null?

Comment (dcposch) [reply packages/daimo-api/src/contract/SwapClogMatcher.ts:68]: good call, S^2 does that now

**Comment by dcposch** [comment]
image

**Comment by andrewliu08** [comment]
should we also test remove and create account?

**Review by andrewliu08 — COMMENTED** [review]
lgtm if everything was tested

**Comment by dcposch** [comment]
* Tested add device + remove device
* V1>V2 upgrade over in this PR: #1224

**Pull request merged**